src/components/PathwayCourse/MiscellaneousCourses/index.js

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { actions as courseActions } from "../../Course/redux/action";
+import { actions as pathwayActions } from "../../PathwayCourse/redux/action";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import { breakpoints } from "../../../theme/constant";
+import { Link } from "react-router-dom";
+import { PATHS, interpolatePath } from "../../../constant";
+import {
+  Container,
+  Box,
+  Grid,
+  Card,
+  CardContent,
+  Typography,
+} from "@mui/material";
+import useStyles from "../styles";
+
+function MiscellaneousCourses() {
+  const dispatch = useDispatch();
+  const { data } = useSelector(({ Course }) => Course);
+  const pathway = useSelector((state) => state.Pathways);
+  const classes = useStyles();
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+
+  useEffect(() => {
+    dispatch(courseActions.getCourses());
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(pathwayActions.getPathways());
+  }, [dispatch]);
+
+  const pathwayCourseId = [];
+  pathway.data &&
+    pathway.data.pathways.filter((pathway) => {
+      pathway.courses.filter((course) => {
+        pathwayCourseId.push(course.id);
+        return course.id;
+      });
+    });
```

Comment:
Variable should probably be renamed `pathwayCourseIds` since it can contains a list of multiple ids although I'm keeping it singular in the suggestion for now.  Also, you don't want to use `filter` here as you're not using its filtered Array `return` value.
```suggestion
  const pathwayCourseId = 
      pathway.data &&
      pathway.data.pathways.map((pathway) => pathway.courses).flat().map(course => course.id) ||
      [];
```